File: 693. binary-number-with-alternating-bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        int x = n ^ (n >> 1);
        return (x & (x + 1)) == 0;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1:
        int n1 = 5;
        boolean output1 = sol.hasAlternatingBits(n1);
        System.out.println("Example 1: Input: n = " + n1);
        System.out.println("Output: " + output1);
        System.out.println("Expected: true");
        System.out.println();
        
        // Example 2:
        int n2 = 7;
        boolean output2 = sol.hasAlternatingBits(n2);
        System.out.println("Example 2: Input: n = " + n2);
        System.out.println("Output: " + output2);
        System.out.println("Expected: false");
        System.out.println();
        
        // Example 3:
        int n3 = 11;
        boolean output3 = sol.hasAlternatingBits(n3);
        System.out.println("Example 3: Input: n = " + n3);
        System.out.println("Output: " + output3);
        System.out.println("Expected: false");
    }
}

File: 693. binary-number-with-alternating-bits.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        x = n ^ (n >> 1)
        return (x & (x + 1)) == 0

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1:
    n = 5
    output = sol.hasAlternatingBits(n)
    print(f"Example 1: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: true")
    print()
    
    # Example 2:
    n = 7
    output = sol.hasAlternatingBits(n)
    print(f"Example 2: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: false")
    print()
    
    # Example 3:
    n = 11
    output = sol.hasAlternatingBits(n)
    print(f"Example 3: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: false")